local_simulation_server.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python3
+
+# Copyright 1996-2022 Cyberbotics Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Local simulation server."""
+
+import os
+import socket
+import subprocess
+import sys
+
+HOST = ''  # Any host can connect
+PORT = int(sys.argv[1])  # Port to listen on
+
+tcp_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+
+tcp_socket.bind((HOST, PORT))
+tcp_socket.listen()
+while True:
+    print(f'Waiting for connection on port {PORT}...')
+    connection, address = tcp_socket.accept()
+
+    print(f'Connection from {address}')
+    data = connection.recv(1024)
+    cmd = data.decode('utf-8').split(' ')
+
+    world_file = cmd[1]
+    if not os.path.isfile(world_file):
+        message = f'FAIL: The world file \'{world_file}\' doesn\'t exist.'
+        connection.sendall(message.encode('utf-8'))
+        print(message, file=sys.stderr)
+        connection.close()
+        continue
+
+    try:
+        test_webots_cmd = subprocess.Popen([cmd[0]])
+    except FileNotFoundError:
+        print('`webots` command could not be found. Falling back to WEBOTS_HOME environment variable.')
+        if 'WEBOTS_HOME' in os.environ:
+            cmd[0] = os.path.join(os.environ['WEBOTS_HOME'], 'webots')
+        else:
+            print('WEBOTS_HOME is not defined. Falling back to default installation folder `/Applications/Webots.app`')
+            cmd[0] = '/Applications/Webots.app/Contents/MacOS/webots'
+
```

Comment:
I would not rely on `WEBOTS_HOME` to make things simpler here and not specific to macOS:
```suggestion
```